Docs: Update migration guides

kylegach opened the pull request:
## What I did

- Add 6 to 8 guide
- Account for auto-blockers

<img width="971" alt="" src="https://github.com/storybookjs/storybook/assets/486540/ed27189d-2a83-4ce7-b79e-96a8a9e9a288">

## Checklist for Contributors

### Testing

#### Manual testing

1. Follow the steps in the [contributing instructions](https://storybook.js.org/docs/react/contribute/new-snippets#preview-your-work) for this branch, `migration-guide-updates-2`

### Documentation

<!-- Please check (put an "x" inside the "[ ]") the applicable items below to indicate which documentation has been updated. -->

- [x] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [x] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [x] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

<!-- CANARY_RELEASE_SECTION -->

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

<!-- CANARY_RELEASE_SECTION -->

kylegach committed:
Update migration guides

- Add 6 to 8 guide
- Account for auto-blockers

jonniebigodes committed:
Merge branch 'next' into migration-guide-updates-2

kylegach committed:
Update migration guides

- Remove babel breaking change mention

jonniebigodes reviewed (changes requested):
Appreciate you taking the time to put this together and so that we can try and unblock users as much as possible during their transition from 6.x to 8. Left some items for you to look into. Let me know once you've addressed them and we'll take it from there.

kylegach committed:
Remove `*.stories.mdx` manual migration in 6 -> 8

- Per: https://github.com/storybookjs/storybook/pull/26201

kylegach commented on docs/migration-guide/from-older-version.md:
```diff
@@ -0,0 +1,179 @@
+
+## Manual migrations
+
+In addition to the automated upgrades above, there are manual migrations that might be required to get Storybook 8 working in your project. We’ve tried to minimize this list to make it easier to upgrade. These include:
```
I disagree? "to get Storybook 8 working in your project" vs. "to get Storybook 8 working on your project". The former is correct grammer.

Also we use the "in" convention in other places, e.g.:

> Installs and initializes the specified version (e.g., @latest, @8, @next) of Storybook into your project

— https://storybook.js.org/docs/api/cli-options#init

jonniebigodes replied:
As we're trying to convey the message that Storybook 8 needs to operate or function properly with the user's project, I'm pretty sure that it should be "on". Since this suggests compatibility or working alongside the project.

kylegach replied:
in : into :: on : onto. We don't say to install Storybook onto your project, we say into. Thus, we should say "in" here.

Here's [another example](https://storybook.js.org/blog/storybook-7-6/) where we use "into":

> Alternatively, for a fresh install, bootstrap Storybook into an existing app with:

I'm not sure how else to explain that "in" is more grammatically and semantically correct here. 🤷‍♂️

kylegach committed:
Apply suggestions from code review

Co-authored-by: jonniebigodes <joaocontadesenvolvimento@gmail.com>

kylegach commented on docs/migration-guide/from-older-version.md:
Turns out, `*.stories.mdx` will be handled by an automigration, so this section is removed entirely.

kylegach committed:
Address comments

- Fix typos and small prose improvements

kylegach committed:
Further prose tweaks

valentinpalkovic commented on docs/migration-guide/from-older-version.md:
```diff
@@ -0,0 +1,158 @@
+
+If any of these changes apply to your project, please read through the linked migration notes before continuing.
+
+If any of these new requirements or changes are blockers for your project, we recommend [migrating to Storybook 7](../../release-7-6/docs/migration-guide).
```
I think this isn't true for Webpack4. As far as I know, we don't have Webpack4 support in Storybook 7.

valentinpalkovic commented on docs/migration-guide/from-older-version.md:
```diff
@@ -0,0 +1,158 @@
+
+In addition to the automated upgrades above, there are manual migrations that might be required to get Storybook 8 working in your project. We’ve tried to minimize this list to make it easier to upgrade. These include:
+
+### Upgrade MDX1 to MDX3
+
+Storybook 8 uses MDX3 by default for rendering [docs](./writing-docs/index.md). The upgrade from MDX1 to MDX3 is not fully automated due to the large number of changes between versions (particularly v1 to v2). Fortunately, we have some tips to help make it a lot easier.
```
This is already part of the automigrations. A manual migration afterwards is only necessary if the user declines the automigration. Nevertheless, the whole `Upgrade MDX1 to MDX3` section adds so much valuable information that I think it's nice to have it!

valentinpalkovic commented on docs/migration-guide/from-older-version.md:
If we detect that the user uses `storyStoryV7: false`, we block the upgrade to 8. And the `storiesOf` API ONLY works with `storyStoryV6 (storyStoryV7: false)`. So in theory, the user has to remove `storyStoryV7: false` to allow the upgrade. In this middle state, Storybook will be broken for user's who continue to use `storiesOf` in Storybook 6. So they upgrade to Storybook 8 and then they have to run the necessary steps to get rid of the `storiesOf` API. I just wanted to clarify this.

kylegach committed:
Address feedback

- Restructure to more accurately provide `storyStoreV7`, `storiesOf`, and MDX 2 guidance

kylegach commented:
@valentinpalkovic — Thank you for the well-considered feedback! I tried to [address your comments here](https://github.com/storybookjs/storybook/pull/26182/commits/5263629a23705db37de22de6cc44acf42bf0e221). Mind taking another look?

kylegach committed:
Merge branch 'next' into migration-guide-updates-2

* next: (39 commits)
  Typo
  cleanup
  Add story for empty argTypes and address review
  fix issues with types, change the comment to not be jsdoc
  Add E2E test for multiple CSF files with same title in autodocs
  Fix overflow bug by using a better link
  remove deprecation of `manager-api`'s `types` export
  Don't show empty arg tables in doc pages
  Fix Yarn2Proxy findInstallations method
  revert component values back to array
  done
  perf(manager): improve performance when switching stories
  Add todo comment
  Refactor MDX to CSF transform function to return an object instead of a tuple
  Rename broken MDX files to .mdx.broken
  Upgrade migration-guide to mention limitations of mdx-to-csf codemod
  Fix issues where stories.js files were generated although a story doesn't exist
  Rename transformed mdx files
  Add version range for removeLegacyMDX1 fix
  Fix typo in automigrate/types.ts
  ...

valentinpalkovic reviewed (approved):
LGTM!
